### ISSUE TITLE
SummaryFileWriter memory leak?

### ISSUE THREAD

**Issue by taylorhansen** [tensorflow/tfjs#7432]
**System information**
- OS Platform and Distribution: Linux Ubuntu 20.04, Node version 16.19.1
- TensorFlow.js installed from: `npm install @tensorflow/tfjs-node`
- TensorFlow.js version: 4.2.0

**Describe the current behavior**

Node memory usage (measured by `process.memoryUsage.rss()`) increases with each SummaryFileWriter call even when properly disposing tensors, as measured by the `tf.memory()` call's return value not changing at all.

The provided JS code steadily goes up to 8 GB before exiting on my machine. When the TF scalar is replaced with a larger tensor (using `writer.histogram()`), memory usage is unchanged. However, when the `writer.scalar()` call is commented out, the memory usage stays constant. There's also little change when running without `writer.flush()` or when using Node's `--expose-gc` flag.

**Describe the expected behavior**

The memory usage shouldn't be growing so quickly. In long-running training scripts that call into SummaryFileWriter multiple times per epoch (e.g. loss/metrics, weight histograms, etc), this can cause excessive memory growth and can possibly invoke Linux's OOM killer if left running too long. My only option now is to limit the number of writer calls.

**Standalone code to reproduce the issue**

```js
// test.js
const tf = require("@tensorflow/tfjs-node");

const writer = tf.node.summaryFileWriter(__dirname);

console.log("step,bytes,type");
for (let i = 0; i < 10_000_000; ++i) {
    const tensor = tf.scalar(0);
    writer.scalar("tensor", tensor, i);
    tensor.dispose();

    if (i % 10_000 === 0) {
        writer.flush();
        global.gc?.();
        const mem = process.memoryUsage();
        console.log(`${i},${mem.rss},rss`);
        console.log(`${i},${mem.heapUsed},heapUsed`);
        console.log(`${i},${mem.heapTotal},heapTotal`);
        console.log(`${i},${mem.external},external`);
        console.log(`${i},${mem.arrayBuffers},arrayBuffers`);
        console.log(`${i},${tf.memory().numBytes},tf`);
    }
}

```

```sh
node --expose-gc test.js > data.csv
```

**Other info / logs**

Attached: [data.csv](https://github.com/tensorflow/tfjs/files/10855587/data.csv)

Plotted:
![image](https://user-images.githubusercontent.com/13547043/222004854-1d68df44-2273-4f9d-a4e9-e5ca70d1c426.png)

The actual summary file comes out to 500MB. Wasn't able to attach due to size.

Plot without `writer.scalar()` call:
![image](https://user-images.githubusercontent.com/13547043/222005426-25cd8ce9-dde0-44c5-838e-903da9e59f50.png)

I'm not sure whether SummaryFileWriter was meant to be called this often, but it's still an easy enough mistake to make that I figured it might be worth posting an issue about it. Any fix would be appreciated.

**Comment by pyu10055** [comment]
thank you for reporting @taylorhansen , we will look into this issue.
